File: Python/Binary Trees/L_levelOrderTraversal.py
import collections


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root):
        q = collections.deque([root])
        res = []
        if not root:
            return res
        while q:
            lvl=[]
            for i in range(len(q)):
                curr = q.popleft()
                if curr.left:
                    q.append(curr.left)
                if curr.right:
                    q.append(curr.right)
                lvl.append(curr.val)
            res.append(lvl)
        return res


    def levelOrder(self, root):
        ans = []
        def recurse(node, level):
            if not node: return
            if level > len(ans): ans.append([])
            ans[level-1].append(node.val)
            recurse(node.left, level+1)
            recurse(node.right, level+1)
        recurse(root, 1)
        return ans

File: cpp/Binary Tree/L_levelOrderTraversal.cpp
#include<bits/stdc++.h>
using namespace std;


struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 };

vector<vector<int>> levelOrder(TreeNode* root) {
    queue<TreeNode*> q;
    q.push(root);
    vector<vector<int>> res;
    if (!root) return res;
    while (q.size()) {
        int siz = q.size();
        vector<int> lvl = {};
        for(int i=0;i<siz;i++) {
            TreeNode* curr = q.front();
            q.pop();
            if (curr->left) q.push(curr->left);
            if (curr->right) q.push(curr->right);
            lvl.push_back(curr->val);
        }
        res.push_back(lvl);
    }
    return res;
}

int main() {

}